### PR TITLE
Fix regression with TPC-H benchmark

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

Closes https://github.com/apache/datafusion-ballista/issues/1058

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Review by Dandandan — APPROVED** [review]

File: ballista/scheduler/src/state/execution_graph/execution_stage.rs [hunk 1]
```diff
@@ -350,9 +349,10 @@ impl UnresolvedStage {
         )?;
 
         // Optimize join order and statistics based on new resolved statistics
-        let optimize_join = JoinSelection::new();
-        let config = SessionConfig::default();
-        let plan = optimize_join.optimize(plan, config.options())?;
+        // let optimize_join = JoinSelection::new();
+        // let config = SessionConfig::default();
+        // let plan = optimize_join.optimize(plan, config.options())?;
```
Comment (viirya) [inline ballista/scheduler/src/state/execution_graph/execution_stage.rs:354]: If not needed, maybe just remove them?

**Review by viirya — APPROVED** [review]

File: ballista/scheduler/src/state/execution_graph/execution_stage.rs (hunk 1 above)
Comment (andygrove) [reply ballista/scheduler/src/state/execution_graph/execution_stage.rs:356]: We do want to enable this again once https://github.com/apache/datafusion/issues/10978 is fixed in DataFusion. I added a comment with a link to this issue

**Comment by andygrove** [comment]
Thanks for the reviews @Dandandan and @viirya

**Comment by my-vegetable-has-exploded** [comment]
I am not sure that https://github.com/apache/datafusion/pull/12967 can solve the problem or not？ 
I will have a try later.

**Pull request merged**